spyder/app/tests/test_mainwindow.py

```diff
@@ -23,11 +62,125 @@ def main_window():
+    assert_array_equal(shell.get_value('arr'), np.array([1, 2, 3]))
+
+    reset_run_code(qtbot, shell, code_editor, nsb)
+
+    # ---- Run lines ----
+    # Run the whole file line by line
+    for _ in range(code_editor.blockCount()):
+        qtbot.keyClick(code_editor, Qt.Key_F9)
+        qtbot.wait(100)
+
+    # Wait until all objects have appeared in the variable explorer
+    qtbot.waitUntil(lambda: nsb.editor.model.rowCount() == 3, timeout=1500)
+
+    # Verify result
+    assert shell.get_value('a') == 10
+    assert shell.get_value('li') == [1, 2, 3]
+    assert_array_equal(shell.get_value('arr'), np.array([1, 2, 3]))
+
+    reset_run_code(qtbot, shell, code_editor, nsb)
+
+    # ---- Run cell and advance ----
+    # Run the three cells present in file
+    for _ in range(3):
+        qtbot.keyClick(code_editor, Qt.Key_Return, modifier=Qt.ShiftModifier)
+        qtbot.wait(100)
+
+    # Wait until all objects have appeared in the variable explorer
+    qtbot.waitUntil(lambda: nsb.editor.model.rowCount() == 3, timeout=1500)
+
+    # Verify result
+    assert shell.get_value('a') == 10
+    assert shell.get_value('li') == [1, 2, 3]
+    assert_array_equal(shell.get_value('arr'), np.array([1, 2, 3]))
+
+    reset_run_code(qtbot, shell, code_editor, nsb)
+
+    # ---- Run cell ----
+    # Run the first cell in file
+    qtbot.keyClick(code_editor, Qt.Key_Return, modifier=Qt.ControlModifier)
+
+    # Wait until the object has appeared in the variable explorer
+    qtbot.waitUntil(lambda: nsb.editor.model.rowCount() == 1, timeout=1500)
+
+    # Verify result
+    assert shell.get_value('a') == 10
+
+    # Press Ctrl+Enter a second time to verify that we're *not* advancing
+    # to the next cell
+    qtbot.keyClick(code_editor, Qt.Key_Return, modifier=Qt.ControlModifier)
+    assert nsb.editor.model.rowCount() == 1
+
+    main_window.close()
+
+
+@pytest.mark.skipif(os.name == 'nt', reason="It's timing out sometimes on Windows")
+def test_open_files_in_new_editor_window(main_window, qtbot):
+    """
+    This tests that opening files in a new editor window
+    is working as expected.
+
+    Test for issue 4085
+    """
+    # Wait until the window is fully up
+    shell = main_window.ipyconsole.get_current_shellwidget()
+    qtbot.waitUntil(lambda: shell._prompt_html is not None, timeout=15000)
+
+    # Set a timer to manipulate the open dialog while it's running
+    QTimer.singleShot(2000, lambda: open_file_in_editor(main_window,
+                                                        'script.py',
+                                                        directory=LOCATION))
+
+    # Create a new editor window
+    # Note: editor.load() uses the current editorstack by default
+    main_window.editor.create_new_window()
+    main_window.editor.load()
+
+    # Perform the test
+    # Note: There's always one file open in the Editor
+    editorstack = main_window.editor.get_current_editorstack()
+    assert editorstack.get_stack_count() == 2
```

Comment:
It's meant to be run in CIs. We could skip it if we're not running it not in a CI.